### DISCUSSION TITLE
What are some things I need to be aware of before installing this on a steam deck?

### DISCUSSION THREAD

**Discussion by dnkmmr69420** [Jovian-Experiments/Jovian-NixOS#178] [Q&A] [answered]
I have a few questions

- Does this require an external keyboard?
- Is the steam ui mode going to run fine on actual deck hardware?
- Is there an onscreen keyboard already preinstalled? (on kde wayland)
- Any major quirks that I need to know about?
- How does that fetch tarball thing at the bottom of the read me work? I coppied and pasted it into a fetch tarball file that is imported and it gave me errors.
- What is the prefered way of getting the config files to the device? Fetchtarball? Git clone and copy and paste? or flakes? (flakes didn't work properly when tested in a vm)
- Are there going to be any major problems happen when the steam deck stuff gets updated upstreem
- Will there ever be installation ISOs that you can just install like regular NixOS but installs jovianNixOS instead?

**Comment by samueldr** [comment, marked as answer]
>   * Does this require an external keyboard?

Unsure. I don't know if the NixOS graphical installer provides an on-screen keyboard.

Nothing really that would differ from a tablet form-factor non-steam-deck computer without a keyboard.

>   * Is the steam ui mode going to run fine on actual deck hardware?

Yes.

>   * Is there an onscreen keyboard already preinstalled? (on kde wayland)

No.

Nothing about the non-steam parts differ from upstream NixOS defaults.

>   * Any major quirks that I need to know about?

Not that I know of

>   * How does that fetch tarball thing at the bottom of the read me work? I coppied and pasted it into a fetch tarball file that is imported and it gave me errors.

It fetches the repo tarball during evaluation, which is intrinsically extracted, and is replaced by the path to the extracted content. In turn this means it imports the modules folder of this repository from your local copy.

Note that without stating what the errors are, we can't say what the issue is. It's possible, for example, that you did not replace the revision with the current revision, or that the error is about the hash not having been updated accordingly.

>   * What is the prefered way of getting the config files to the device? Fetchtarball? Git clone and copy and paste? or flakes? (flakes didn't work properly when tested in a vm)

It depends on the user's own preferences.

>   * Are there going to be any major problems happen when the steam deck stuff gets updated upstreem


That is an unknowable. But we are already tracking what the vendor is doing closely, and before it hits their stable release branch, so our track record is really good.

Rarely we even fixed and handled issues they introduced in yet-to-be-published images.

>   * Will there ever be installation ISOs that you can just install like regular NixOS but installs jovianNixOS instead?

Maybe. It's on the radar, but low priority.

**Comment by dnkmmr69420** [comment]
Ah thank you for the answers. I am gonna make a pull request for the documentation that has a link to this issue.

**Comment by ghost** [comment]
High Speed SD card is basically required. Tried using a non class 10 sd card any thing else can take hours to boot from sd

**Comment by samueldr** [comment]
> High Speed SD card is basically required. Tried using a non class 10 sd card any thing else can take hours to boot from sd

Not if you install to the internal storage! (assuming you were talking about the install target.)
